tests/test_install_deps.py

```diff
@@ -23,42 +25,33 @@ def test_resolve_dependencies_install_deps__via_local_cache(local_pypi):
     }
 
 
-def test_resolve_dependencies_install_deps__handle_pip_install_failure(
+def test_resolve_dependencies_install_deps__raises_unresolved_error_on_pip_install_failure(
     caplog, local_pypi
 ):
-    # TemporaryPipInstallResolver will handle the inevitable pip install error
-    # and return to resolve_dependencies() with the missing package unresolved.
-    # For now, IdentityMapping "saves the day" and supplies a Package object.
-    # Soon, this should result in an unresolved package error instead.
+    # This tests the case where TemporaryPipInstallResolver encounters the
+    # inevitable pip install error and returns to resolve_dependencies()
+    # with the missing package unresolved.
+    # Since either install_deps or IdentityMapping are the final resolvers,
+    # this should raise an `UnresolvedDependenciesError`.
     caplog.set_level(logging.WARNING)
-    actual = resolve_dependencies(
-        ["does_not_exist"], setup_resolvers(install_deps=True)
-    )
-    assert actual == {
-        "does_not_exist": Package(
-            "does_not_exist", {"does_not_exist"}, IdentityMapping
-        ),
-    }
+
+    with pytest.raises(UnresolvedDependenciesError):
+        resolve_dependencies(["does_not_exist"], setup_resolvers(install_deps=True))
+
     assert all(word in caplog.text for word in ["pip", "install", "does_not_exist"])
 
 
-def test_resolve_dependencies_install_deps__pip_install_some_packages(
+def test_resolve_dependencies_install_deps_on_mixed_packages__raises_unresolved_error(
```

Comment:
In my comment about `UnresolvedDependenciesError`, here is the kind of test I was thinking about. This verifies that we preserve the nuance of what packages actually failed in our error messaging:

```suggestion
def test_resolve_dependencies_install_deps__unresolved_error_only_warns_failing_packages(
    caplog, local_pypi
):
    # When we fail to install _some_ - but not all - packages, the error message
    # should only mention the packages that we failed to install.
    caplog.set_level(logging.WARNING)
    deps = {"click", "does_not_exist", "leftpadx"}

    with pytest.raises(UnresolvedDependenciesError):
        resolve_dependencies(deps, setup_resolvers(install_deps=True))

    assert "Failed to install 'does_not_exist'" in caplog.text
    assert "Failed to install 'click'" not in caplog.text
    assert "Failed to install 'leftpadx'" not in caplog.text


def test_resolve_dependencies_install_deps_on_mixed_packages__raises_unresolved_error(
```

Reply:
I somehow missed this comment. Just committed the change. Will merge once the CI passes.